adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m

```diff
@@ -13,47 +13,92 @@
 // limitations under the License.
 
 #import "GADMediationAdapterPangle.h"
-
 #import "GADMediationAdapterPangleConstants.h"
+#import <BUAdSDK/BUAdSDK.h>
+#import "GADPangleRTBBannerRenderer.h"
+#import "GADPangleRTBInterstitialRenderer.h"
+#import "GADPangleRTBRewardedRenderer.h"
+#import "GADMediationAdapterPangleConstants.h"
+#import "GADMAdapterPangleUtils.h"
 
-@implementation GADMediationAdapterPangle
+static NSInteger _coppa = -1,_gdpr = -1, _ccpa = -1;
+
+@implementation GADMediationAdapterPangle {
+    /// Pangle banner ad wrapper.
+    GADPangleRTBBannerRenderer *_bannerRenderer;
+    /// Pangle interstitial ad wrapper.
+    GADPangleRTBInterstitialRenderer *_interstitialRenderer;
+    /// Pangle rewarded ad wrapper.
+    GADPangleRTBRewardedRenderer *_rewardedlRenderer;
+}
 
 - (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
                          completionHandler:(nonnull GADRTBSignalCompletionHandler)completionHandler {
-  // TODO: Start signal generation and call completionHandler when the signal generation finishes or fails.
-  completionHandler("", nil);
+    NSString *signals = [BUAdSDKManager getBiddingToken:nil];
+    if (completionHandler) {
+        completionHandler(signals,nil);
+    }
 }
 
 + (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
              completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Pangle ad network SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
+    NSMutableSet *appIds = [[NSMutableSet alloc]init];
```

Comment:
Use `GADMAdapterPangleMutableSetAddObject` as in https://github.com/googleads/googleads-mobile-ios-mediation/pull/351/files#r819437507.